velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -264,6 +266,40 @@ struct LastDayFunction {
   }
 };
 
+namespace {
+// clang-format off
+static const std::unordered_map<std::string, int> kWeekNum = {
+    {"SU", 0}, {"SUN", 0}, {"SUNDAY", 0},
+    {"MO", 1}, {"MON", 1}, {"MONDAY", 1},
+    {"TU", 2}, {"TUE", 2}, {"TUESDAY", 2},
+    {"WE", 3}, {"WED", 3}, {"WEDNESDAY", 3},
+    {"TH", 4}, {"THU", 4}, {"THURSDAY", 4},
+    {"FR", 5}, {"FRI", 5}, {"FRIDAY", 5},
+    {"SA", 6}, {"SAT", 6}, {"SATURDAY", 6}
+};
+// clang-format on
+} // namespace
+
+template <typename T>
+struct NextDayFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE bool call(
+      out_type<Date>& result,
+      const arg_type<Date>& date,
+      const arg_type<Varchar>& dayOfWeek) {
+    auto dateTime = getDateTime(date);
+
+    auto iter = kWeekNum.find(boost::to_upper_copy<std::string>(dayOfWeek));
```

Comment:
Can we use `boost::to_upper` to replace current `boost::to_upper_copy`? @mbasmanova The modification of the string is done in-place if to_upper() is used.

Reply:
I think we should not change the input as if it's used by others.